Source/JavaScriptCore/bytecode/BytecodeList.rb

```diff
@@ -1044,6 +1044,53 @@
         operand: :offset,
     }
 
+op :resolve_and_get_from_scope,
+    args: {
+        dst: VirtualRegister,
+        scope: VirtualRegister,
```

Comment:
Why do we need this operand anymore? The whole point of switching to tmps is so that we don't define a scope anymore. And we can do this if we only convert places in the bytecode generator where we can prove that the result of the resolve_scope is only used by the get_from_scope.

Reply:
I think we cannot remove this `scope` operand. Here, the `scope` is the input for resolve_scope.

```
[  28] resolve_scope       dst:loc10, scope:loc4, var:1, resolveType:GlobalProperty, localScopeDepth:0
[  35] get_from_scope     dst:loc7, scope:loc10, var:1, getPutInfo:2048<...>, localScopeDepth:0, offset:0
```

Please check the emitted bytecode for resolve_scope, which needs a `dst` and a `scope`. Since the destination of resolve_scope might be the input `scope` for get_from_scope only, we can introduce our new opcode resolve_and_get_from_scope by getting rid of the `dst` of resolve_scope (`dst:loc10`) and `scope` of get_from_scope (`scope:loc10`).

...
[  28] resolve_and_get_from_scope dst:loc7, scope:loc4, resolvedScope:loc10, var:1, resolveType:GlobalProperty, getPutInfo:2048<...>, localScopeDepth:0, offset:0
...

However, we still need `scope:loc4` to get the resolved scope and `dst:loc7`as the destination of get_from_scope. That's why we cannot remote this `scope` operand in resolve_and_get_from_scope unless we have an approach to resolve scope without the input `scope`.

Reply:
ignore me, I wasn't thinking right. This is needed as an input